### PR TITLE
[12.0][FIX] web_widget_one2many_product_picker: Use variant image

### PR DESCRIPTION
**Pull request by Tardo**
Ensures the usage of the image variant if exists.

cc @tecnativa TT27554

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

File: web_widget_one2many_product_picker/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Tecnativa - Alexandre D. Díaz
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models, tools
+
+
+class ProductProduct(models.Model):
+    _inherit = ['product.product']
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]:
```suggestion
    _inherit = 'product.product'
```

File: web_widget_one2many_product_picker/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    image_variant_medium = fields.Binary(
+        "Variant Image Medium (Computed)",
+        compute='_compute_variant_image',
+        help="This field holds the image used as image for the product variant"
+             "or produdct image medium, limited to 512x512px.",
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]:
```suggestion
             "or product image medium, limited to 512x512px.",
```

File: web_widget_one2many_product_picker/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    )
+
+    image_variant_big = fields.Binary(
+        "Variant Image Big (Computed)",
+        compute='_compute_variant_image',
+        help="This field holds the image used as image for the product variant"
+             "or produdct image, limited to 1024x1024px.",
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]:
```suggestion
             "or product image, limited to 1024x1024px.",
```

File: web_widget_one2many_product_picker/models/product_product.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    )
+
+    @api.one
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]: Don't use this deprecated form

File: web_widget_one2many_product_picker/models/product_product.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    @api.depends('image_variant', 'product_tmpl_id.image')
+    def _compute_variant_image(self):
+        if self.image_variant:
+            resized_images = tools.image_get_resized_images(
+                self.image_variant,
+                return_big=True,
+                avoid_resize_medium=False)
+            self.image_variant_medium = resized_images['image_medium']
+            self.image_variant_big = resized_images['image']
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]: Maybe?
```suggestion
            self.image_variant_big = self.image_variant
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
Updated

File: web_widget_one2many_product_picker/models/product_product.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    )
+
+    @api.depends('image_variant', 'product_tmpl_id.image')
+    def _compute_variant_image(self):
+        self.ensure_one()
+        if self.image_variant:
+            resized_images = tools.image_get_resized_images(
+                self.image_variant,
+                return_big=False,
+                avoid_resize_medium=False)
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]: If you are avoiding this, isn't the same as assigning `self.image_variant` to both?

File: web_widget_one2many_product_picker/models/product_product.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    image_variant_big = fields.Binary(
+        "Variant Image Big (Computed)",
+        compute='_compute_variant_image',
+        help="This field holds the image used as image for the product variant"
+             "or product image, limited to 1024x1024px.",
+    )
+
+    @api.depends('image_variant', 'product_tmpl_id.image')
+    def _compute_variant_image(self):
+        self.ensure_one()
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/product_product.py]: You have removed `@api.one`, but this method can be accessed by multi-recordset. You can't restrict it this way, but do here the `for record in self` loop.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: web_widget_one2many_product_picker/models/product_product.py (hunk 6 above)
Comment (Tardo) [reply web_widget_one2many_product_picker/models/product_product.py]: No, but i change it to True.
This parameter is used to know if need resize the image when is small.

**Comment by Tardo** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**